src/pudl/transform/eia.py

```diff
@@ -583,82 +587,59 @@ def _boiler_generator_assn(  # noqa: C901
     Args:
         eia_transformed_dfs: a dictionary of post-transform dataframes
             representing the EIA database tables.
-        eia923_years: a list of the years of EIA 923 data that
-            should be used to infer the boiler-generator associations. By
-            default it is all the working years of data.
-        eia860_years: a list of the years of EIA 860 data that
-            should be used to infer the boiler-generator associations. By
-            default it is all the working years of data.
-        debug: If True, include columns in the returned dataframe
-            indicating by what method the individual boiler generator
-            associations were inferred.
 
     Returns:
-        eia_transformed_dfs (dict): Returns the same dictionary of dataframes
-        that was passed in, and adds a new dataframe to it representing
-        the boiler-generator associations as records containing
-        plant_id_eia, generator_id, boiler_id, and unit_id_pudl
+        A dataframe containing the boiler generator associations.
 
     Raises:
-        AssertionError: If the boiler - generator association graphs are not
-            bi-partite, meaning generators only connect to boilers, and boilers
-            only connect to generators.
-        AssertionError: If all the boilers do not end up with the same unit_id
-            each year.
-        AssertionError: If all the generators do not end up with the same
-            unit_id each year.
+        AssertionError: If the boiler - generator association graphs are not bi-partite,
+            meaning generators only connect to boilers, and boilers only connect to
+            generators.
+        AssertionError: If all the boilers do not end up with the same unit_id each
+            year.
+        AssertionError: If all the generators do not end up with the same unit_id each
+            year.
     """
+    debug = context.op_config["debug"]
+    eia_settings = context.resources.dataset_settings.eia
+    eia923_years = eia_settings.eia923.years
+    eia860_years = eia_settings.eia860.years
+
     if eia923_years is None:
         eia923_years = DataSource.from_id("eia923").working_partitions["years"]
     if eia860_years is None:
         eia860_years = DataSource.from_id("eia860").working_partitions["years"]
```

Comment:
this may be oos but is this still necessary? doesn't it default to using the working_partitions now?

Reply:
Hmm. I'm not sure how it works now.  If you try and create an `EiaSettings` class without any eia860/923 years specified (i.e. if you tell it not to do any EIA data processing) what does Dagster do now? And could it ever get to this function?

I think this may be a holdover from long before we had anything resembling the settings. Probably this can/should just depend directly on the settings object instead of these lists of years.  @bendnorman what do you think?

Reply:
Switch to taking a settings object as input directly.